packages/lesswrong/components/tagging/TagFlagItem.tsx

```diff
@@ -35,38 +36,62 @@ const styles = (theme: ThemeType): JssStyles => ({
   }
 })
 
-const TagFlagItem = ({documentId, showNumber = true, allPages = false, style = "grey", classes }: {
+
+const TagFlagItem = ({documentId, itemType = "tagFlagId", showNumber = true, style = "grey", classes }: {
   documentId?: string,
+  itemType?: "tagFlagId" | "allPages" | "userPages" 
   showNumber?: boolean,
-  allPages?: boolean, 
+  allPages?: boolean,
+  myPages?: boolean,
   style?: "white"|"grey"|"black",
   classes: ClassesType,
 }) => {
   const { LWPopper, ContentItemBody } = Components;
   const {eventHandlers, hover, anchorEl, stopHover } = useHover();
+  const currentUser = useCurrentUser();
   const { document: tagFlag } = useSingle({
     documentId,
     collection: TagFlags,
     fetchPolicy: "cache-first",
     fragmentName: "TagFlagFragment",
   })
-  const multiTerms = allPages ? {view: "allPagesByNewest"} : { view: "tagsByTagFlag", tagFlagId: tagFlag?._id}
+  
+  
+  const getTagFlagItemTerms = (itemType) => { 
+    switch(itemType) {
+      case 'allPages':
+        return {view: "allPagesByNewest"}
+      case 'userPages':
+        return {view: "userTags", userId: currentUser?._id}
+      default: //tagFlagId type
+        return {view: "tagsByTagFlag", tagFlagId: tagFlag?._id}
+    }
+  }
+    
   const { totalCount, loading } = useMulti({
-    terms: multiTerms,
+    terms: getTagFlagItemTerms(itemType),
     collection: Tags,
     fragmentName: "TagWithFlagsFragment",
     limit: 0,
     skip: !showNumber,
     enableTotal: true
   });
+  
   const rootStyles = classNames(classes.root, {[classes.black]: style === "black", [classes.white]: style === "white"});
-  const tagFlagDescription = allPages ? "All Pages" : `tagFlag ${tagFlag?._id}`
-  const tagFlagText = allPages ? "All Wiki-Tags" : tagFlag?.name
-  const innerHTML = {
-    __html: allPages
-      ? "All Wiki-Tags sorted by most recently created, including those with no flags set."
-      : tagFlag?.contents?.html || "" 
-  }
+  
+  var tagFlagDescription = `tagFlag ${tagFlag?._id}`
+  var tagFlagText = tagFlag?.name
+  var innerHTML = tagFlag?.contents?.html || ""
+  
+  if (itemType === "allPages") {
+    var tagFlagDescription = "All Pages"
+    var tagFlagText = "All Wiki-Tags"
+    var innerHTML = "All Wiki-Tags sorted by most recently created, including those with no flags set."
```

Comment:
This is quite bad, and I don't actually know the behavior. My guess is this just doesn't work, because you aren't actually reassigning the variable `innerHTML`, you are creating a new variable in the scope that will be dismissed as soon as you leave the current closure. 

This should just be `innerHTML = ...` without the leading `var`